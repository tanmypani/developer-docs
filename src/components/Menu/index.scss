@import '../../assets/styles/variables.scss';

.menuWrapper {
    display: flex;
    flex: 1;
    padding-left: 20px;

    .menu {
        float: left;
        overflow: hidden;
        color: $lightgrey;
        padding: 5px 20px 5px 0;
    }

    .menu .menubtn {
        font-size: $font-size-normal;
        border: none;
        outline: none;
        color: $white;
        background-color: inherit;
        margin: 0;
        padding-bottom: 14px;
    }

    .menuContent {
        display: none;
        position: absolute;
        background-color: var(--body-background-color);
        min-width: 74px;
        box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
        z-index: 1;
    }

    .menuContent > div {
        float: none;
        cursor: pointer;
        color: var(--primary-color);
        padding: 8px 16px;
        text-decoration: none;
        display: block;
        text-align: left;
    }

    .menuContent div:hover {
        color: var(--secondary-color);
    }

    .menu:hover .menuContent {
        display: block;
    }

    .arrowDown {
        height: 0.8rem;
        width: 0.8rem;
        vertical-align: middle;
        margin-left: 5px;
    }
    .subLabel {
        font-size: $font-size-small;
        font-weight: $font-weight-normal;
        padding: 3px 0px;
    }
}
