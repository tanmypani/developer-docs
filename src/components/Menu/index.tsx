import React, { useState, useEffect } from 'react';
import { AiOutlineCaretDown } from '@react-icons/all-files/ai/AiOutlineCaretDown';
import './index.scss';

const Menu = (props: { config: {} }) => {
    const { config } = props;
    const handelClick = (menu: { link: string; external: boolean }) => {
        const { external = false, link } = menu;
        let url = location.origin + '/' + link;
        let target = '_self';
        if (external) {
            url = link;
            target = '_blank';
        }
        window.open(url, target);
    };

    return (
        <div className="d-inline-block headerLink menuWrapper">
            {config.map(({ name, child }) => (
                <div className="menu">
                    <button className="menubtn">{name}</button>
                    <div className="menuContent">
                        {child?.map((d: { label: string; link: string }) => {
                            return (
                                <div
                                    data-testid={`option-${d?.label}`}
                                    onClick={() => handelClick(d)}
                                >
                                    {d?.label}
                                </div>
                            );
                        })}
                    </div>
                </div>
            ))}
        </div>
    );
};

export default Menu;
