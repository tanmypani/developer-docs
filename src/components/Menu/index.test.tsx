import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';

import Menu from './index';

describe('Menu', () => {
    window.open = jest.fn();

    it('should renders correctly', () => {
        const tree = renderer.create(<Menu config={[]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
