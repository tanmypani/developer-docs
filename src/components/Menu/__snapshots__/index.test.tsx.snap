// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should renders correctly 1`] = `
<div
  className="d-inline-block headerLink menuWrapper"
/>
`;
